### PR TITLE
Add China Region S3 URL

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/aws-parallelcluster.cfn.json [hunk 1]
```diff
@@ -2901,27 +2901,20 @@
...
-                "    s3_prefix=s3-",
+                "  _region=",
                 {
                   "Ref": "AWS::Region"
                 },
                 "\n",
+                "if [ \"${custom_cookbook}\" != \"NONE\" ]; then\n",
+                "  cookbook_url=${custom_cookbook}\n",
+                "else\n",
+                "  if [ \"${_region}\" =~ \"^cn-\" ]; then\n",
```
Comment (mauri-melato) [inline cloudformation/aws-parallelcluster.cfn.json]: Mind double `[[ ]]` and no quotes around the regex `if [[ \"${_region}\" =~ ^cn- ]];`

File: cloudformation/aws-parallelcluster.cfn.json [hunk 2]
```diff
@@ -1336,15 +1355,15 @@
         },
         "TemplateURL": {
           "Fn::Sub": [
-            "https://${s3_domain}/${AWS::Region}-aws-parallelcluster/templates/efs-substack-${version}.cfn.json",
+            "https://s3.${AWS::Region}.${s3_url}/${AWS::Region}-aws-parallelcluster/templates/efs-substack-${version}.cfn.json",
             {
-              "s3_domain": {
-                "Fn::If": [
-                  "GovCloudRegion",
+              "s3_url": {
```
Comment (mauri-melato) [inline cloudformation/aws-parallelcluster.cfn.json]: If possible let's define s3_url suffix or better the whole s3_url at the beginning once and then use it with Ref without to repeat the FindInMap construct multiple times.

File: cli/pcluster/pcluster.py [hunk 3]
```diff
@@ -749,13 +749,12 @@ def get_cookbook_url(config, tmpdir):
         return config.args.custom_ami_cookbook
 
     cookbook_version = get_cookbook_version(config, tmpdir)
-    if config.region == "us-east-1":
-        return "https://s3.amazonaws.com/%s-aws-parallelcluster/cookbooks/%s.tgz" % (config.region, cookbook_version)
-
-    return "https://s3.%s.amazonaws.com/%s-aws-parallelcluster/cookbooks/%s.tgz" % (
+    s3_suffix = '.cn' if config.region.startswith('cn') else ''
+    return "https://s3.%s.amazonaws.com%s/%s-aws-parallelcluster/cookbooks/%s.tgz" % (
```
[the remark this answers is not in the document]
Comment (sean-smith) [reply cli/pcluster/pcluster.py]: fixed

File: cli/pcluster/cfnconfig.py [hunk 4]
```diff
@@ -487,16 +487,12 @@ def __init__(self, args):  # noqa: C901 FIXME!!!
...
-                        )
-                    else:
-                        self.template_url = (
-                            "https://s3.%s.amazonaws.com/%s-aws-parallelcluster/templates/"
-                            "aws-parallelcluster-%s.cfn.json" % (self.region, self.region, self.version)
-                        )
+                    s3_suffix = '.cn' if self.region.startswith('cn') else ''
+                    self.template_url = (
+                            "https://s3.%s.amazonaws.com%s/%s-aws-parallelcluster/templates/"
+                            "aws-parallelcluster-%s.cfn.json" % (self.region, self.region, s3_suffix, self.version)
```
[the remark this answers is not in the document]
Comment (sean-smith) [reply cli/pcluster/cfnconfig.py]: fixed

File: cloudformation/aws-parallelcluster.cfn.json (hunk 2 above)
Comment (sean-smith) [reply cloudformation/aws-parallelcluster.cfn.json]: If I do, it needs to be a parameter 😌

**Comment by sean-smith** [comment]
### Testing
* Commercial `sge` ✅
* Commercial `awsbatch` ✅

**Pull request merged**